preprocessors/openstreetmap/osm_service.py

```diff
@@ -154,6 +174,346 @@ def process_streets_data(OSM_data, bbox_coordinates):
         return processed_OSM_data
 
 
+def get_new_nodes(unbounded_node_list, node_list, bbox_coordinates):
+    # node_list1 contains the unbounded nodes
+    # node_list  contains the bounded nodes
+    node_list1 = unbounded_node_list
+    i = len(node_list)
+    j = len(node_list1)
+
+    # Case I: When there is only one point (node) in the node_list,
+    # but more points in the unbounded_node_list, meaning that some
+    # of the street nodes have been excluded
+    # for failing to meet the bounds conditions.
+    if node_list:
+        if i == 1 and j > i:
+            # index gives the location of this
+            # "single point node" in the original list
+            # (i.e., unbounded_node_list).
+            index = node_list1.index(node_list[0])
+            lat1 = node_list[0]["lat"]
+            lon1 = node_list[0]["lon"]
+
+            # If the "single point node" above has both preceding
+            # and succeeding nodes in the original node list
+            # (i.e., in the unbounded_node_list/node_list1).
+            # Then compute the possible points for both the preceding and
+            # succeeding nodes, that will satisfy the bounds conditions.
```

Comment:
Please break this out into flags for if a preceding and succeeding node are necessary, and add each. It seems there is redundant code in the case where both are present.